### DISCUSSION TITLE
S3 API CompleteMultipartUpload

### DISCUSSION THREAD

**Discussion by Thomas-Mart** [seaweedfs/seaweedfs#4746] [General]
Hello

I've a question to the multipart upload S3 API and espacially about the CompleteMultipartUpload part. In the Amazon S3 API reference there are different errors described about the complete step. Here's a list about the errors I mention:

_CompleteMultipartUpload has the following special errors:
- Error code: EntityTooSmall
    - Description: Your proposed upload is smaller than the minimum allowed object size. Each part must be at least 5 MB in size, 
    except the last part.
    - 400 Bad Request

- Error code: InvalidPart
    - Description: One or more of the specified parts could not be found. The part might not have been uploaded, or the specified 
    entity tag might not have matched the part's entity tag.
    - 400 Bad Request

Error code: InvalidPartOrder
    - Description: The list of parts was not in ascending order. The parts list must be specified in order by part number.
    - 400 Bad Request

Error code: NoSuchUpload
    - Description: The specified multipart upload does not exist. The upload ID might be invalid, or the multipart upload might have 
    been aborted or completed.
    - 404 Not Found_ 

Are all these errors supported by seaweedfs or if not which parts are supported?
For me it seems that only the last error code (404 - NoSuchUpload) is supported. 
E.g. I tried to create 3 parts of a file but only uploaded 2 of them. In the CompleteMultipartUpload I've created the XML body listing all 3 parts of the file. So when I try to finish the upload as described I get a 200 response without any error code and the file is created but without the size I expect.
And therefore I've another question. Because of which criteria does seaweedfs decide which parts are part of the complete multipart upload? Is it only because of the 'PutObjectPart' calls or does it really use the XML structure provided in the 'CompleteMultipartUpload'. 

Thanks and best regards,
Thomas

**Comment by chrislusf** [comment]
added code to address your example https://github.com/seaweedfs/seaweedfs/commit/6792db21dc32bff90330945f0dd303bd490517cb
